var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -545,6 +547,22 @@ def cmake_args(self):
                 f'-DcuFFTMp_ROOT={self.spec["nvhpc"].prefix}/Linux_{self.spec.target.family}'
                 + f'/{self.spec["nvhpc"].version}/math_libs'
             )
+            # Add option to include cufft-mp's nvhshmem dependency.
+            # This currently requires the NVIDIA device drivers to be installed on the build
+            # machine to fulfill nvshmem_host's dependency on nvidia-ml.
+            options.append(
+                f'-DcuFFTMp_NVSHMEM_ROOT={self.spec["nvhpc"].prefix}/'
+                + f'Linux_{self.spec.target.family}/{self.spec["nvhpc"].version}'
+                + f'/comm_libs/{self.spec["cuda"].version.up_to(2)}/nvshmem_cufftmp_compat'
```

Comment:
Whether this file even exists depends on the version of cuFFTMp. I found that there were two versions of the 2023 nvhpcsdk that bundled such extra versions of nvshmem, and several that did not. So a robust Spack implementation might be tricky. Simplest is for this Spack recipe to refuse to build with cuFFTMp support from the nvhpc versions that need the `compat` versions.